### PR TITLE
fix: checkout types & feat: customer module

### PR DESCRIPTION
**Pull request by carstenlebek**
Made optional fields optional,  added `subscription_preview` and fixed the type for `custom`

### CODE REVIEW

**Comment by NuroDev** [comment]
Thanks for the work on this @carstenlebek  🙏. I am going to have a full look through the changes shortly.

On quick glance I can see a lot of code formatting changes. I have just update `main` to include a Prettier config & a `format` command to fix any formatting issues. Please pull the changes & run the `format` command so everything can stay consistant.

**Comment by carstenlebek** [comment]
Sorry, my bad 🤦🏼‍♂️ fixed the formatting issues

File: src/modules/checkout/checkout.types.ts [hunk 1]
```diff
@@ -73,6 +73,10 @@ export interface LemonsqueezyCheckoutOptions {
    * If `false`, hide the product media
    */
   media?: boolean;
+  /**
+   * If false, hide the "You will be charged..." subscription preview text
```
Comment (NuroDev) [inline src/modules/checkout/checkout.types.ts:77]:
```suggestion
   * If `false`, hide the "You will be charged..." subscription preview text
```

File: src/modules/customer/customer.action.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import {
```
Comment (NuroDev) [inline src/modules/customer/customer.action.ts:1]:
```suggestion
import type {
```

**Review by NuroDev — CHANGES_REQUESTED** [review]
Overall looks pretty good. A few small things then should be good to merge:

- Minor code suggestions
- Add tests to `src/client/client.test.ts`
- Add the `/customer` export to `package.json`

**Comment by carstenlebek** [comment]
All done 🥳

**Review by NuroDev — APPROVED** [review]

**Pull request merged**